### PR TITLE
nasty way to make versions unique

### PR DESCRIPTION
**Pull request by dbolser-ebi**
could be done better I'm sure

### CODE REVIEW

**Comment by dbolser-ebi** [comment]
One one database tested, the output file was strictly identical, but returned 4 times faster.

File: misc-scripts/assembly/ensembl_assembly_to_chain.pl [hunk 1]
```diff
@@ -160,7 +160,18 @@ sub run_on_dba {
...
+  my %unique_mappings;
+  for (@$mappings){
+      die "Can't parse mapping string for coord system version '$_'!\n"
+          unless /.+:(.+)#.+:(.+)/;
+      my $version_1 = $1;
+      my $version_2 = $2;
+      $unique_mappings{$version_1. ":". $version_2}++;
+  }
+  my @unique_mappings;
+  push @unique_mappings, [split(/:/, $_)]
```
Comment (nerdstrike) [inline misc-scripts/assembly/ensembl_assembly_to_chain.pl]: Seems a little odd to join the versions together and then split them up again.

Comment (dbolser-ebi) [inline misc-scripts/assembly/ensembl_assembly_to_chain.pl]: Quickest way I could think of to make a unique list of pairs... hacky I
know.

Comment (andrewyatz) [inline misc-scripts/assembly/ensembl_assembly_to_chain.pl]: What no schwarzian transform? Boo :)

**Comment by dbolser-ebi** [comment]
my @unique_mappings = List::UtilsBy::uniq_by { join ":", @$_ } @mappings; #?

**Comment by andrewyatz** [comment]
Sorry @dbolser-ebi that this went dead. Using an external module is probably overkill. How about this?

```perl
my %unique_mappings;
foreach my $mapping (@{$mappings}) {
    die "Can't parse mapping string for coord system version '${mapping}'!\n"
       unless /.+:(.+)#.+:(.+)/;
    my $version_1 = $1;
    my $version_2 = $2;
    $unique_mappings{$version_1. ":". $version_2} = [$version_1, $version_2];
}
return [values %unique_mappings];
```

Avoids the need to join and split, avoids use of `$_` and reduces the name clobbering between hash and arrays.

**Comment by dbolser-ebi** [comment]
Looks nice.

**Comment by magaliruffier** [comment]
Can you please incorporate Andy's suggestion and test it still works for you?

**Comment by magaliruffier** [comment]
@dbolser-ebi we'd like to incorporate the change before code branching, which is on Thursday
can you please test Andy's suggestion and update the PR accordingly if that works for you?

**Comment by dbolser-ebi** [comment]
Tested fine, i.e. the MD5SUM of the output files was identical before and after. Didn't observe the pointless looping.

**Comment by magaliruffier** [comment]
Great, thank you

**Pull request merged**